test_conformance/SVM/test_unified_svm_map_unmap.cpp

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright (c) 2025 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "unified_svm_fixture.h"
+#include "harness/conversions.h"
+#include "harness/testHarness.h"
+#include "harness/typeWrappers.h"
+
+struct UnifiedSVMMapUnmap : UnifiedSVMBase
+{
+    using UnifiedSVMBase::UnifiedSVMBase;
+
+    // Test the clEnqueueSVMMap and clEnqueueSVMUnmap functions for random
+    // ranges of a USM allocation and validate the event types.
+    cl_int test_SVMMapUnmap(USVMWrapper<cl_uchar> *mem, cl_map_flags flags)
+    {
+        cl_int err = CL_SUCCESS;
+
+        for (size_t it = 0; it < test_iterations; it++)
+        {
+
+            size_t offset = get_random_size_t(0, alloc_count - 1, d);
+            size_t length = get_random_size_t(1, alloc_count - offset, d);
+
+            void *ptr = &mem->get_ptr()[offset];
+
+            clEventWrapper map_event;
+            err = clEnqueueSVMMap(queue, CL_FALSE, flags, ptr, length, 0,
+                                  nullptr, &map_event);
+            test_error(err, "clEnqueueSVMMap failed");
+
+            clEventWrapper unmap_event;
+            err = clEnqueueSVMUnmap(queue, ptr, 0, nullptr, &unmap_event);
+            test_error(err, "clEnqueueSVMUnmap failed");
+
+            err = clFinish(queue);
+            test_error(err, "clFinish failed");
+
+            err = check_event_type(map_event, CL_COMMAND_SVM_MAP);
+            test_error(err,
+                       "Invalid command type returned for clEnqueueSVMMap");
+
+            err = check_event_type(unmap_event, CL_COMMAND_SVM_UNMAP);
+            test_error(err,
+                       "Invalid command type returned for clEnqueueSVMUnmap");
+        }
+
+        return err;
+    }
+
+    cl_int run() override
+    {
+        cl_int err;
+        cl_map_flags test_flags[] = { CL_MAP_READ, CL_MAP_WRITE,
+                                      CL_MAP_WRITE_INVALIDATE_REGION,
+                                      CL_MAP_READ | CL_MAP_WRITE };
+
+        cl_uint max_ti = static_cast<cl_uint>(deviceUSVMCaps.size());
+
+        for (cl_uint ti = 0; ti < max_ti; ti++)
```

Comment:
This test should only execute for SVM types that have the capability `CL_SVM_CAPABILITY_HOST_MAP_KHR`.

Reply:
will fix